src/lib.rs

```diff
@@ -229,8 +246,8 @@ impl<'spir> PetSpirv<'spir> {
             .filter(|c| match c {
                 '$' | '.' => false,
                 _ => true,
-            }).collect();
-        writeln!(write, "digraph {{");
```

Comment:
You don't have to remove this, just replace `write!(write, "digraph {{")` with this:
```rust
write!(write, "subgraph cluster_{} {{", fn_id);
```
(or `dot_friendly_name` instead of `fn_id` but I doubt that's a good idea - it just has to be unique, anyway)

This will ensure the nodes stay grouped together, you get a border *and* you can use `label = "..."` instead of having a node just to name the function.

We can also use *nested* `subgraph`s to indicate loop/branch regions (I guess I could start working on that separately).

Reply:
I was trying this and thought `subgraph`s default to some unwanted layout, e.g.:

![](http://build.lyken.rs/~eddy/tmp/JbAD.svg)

But it turns out that's just what Graphviz does to the graph itself, `subgraph cluster_...` only added borders to it:

![](http://build.lyken.rs/~eddy/tmp/nj1P.svg)

Reply:
`pack = true;` for the whole `digraph`, and using `subgraph`s, results in this (IMO ideal) output:

![](http://build.lyken.rs/~eddy/tmp/B7Ye/_.svg)